cryostatio/cryostat3
# fix(db): correct use of Transactional annotation

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #405
Fixes: #440

## Description of the change:
Removes useless instances of `Transactional` annotation and replaces them with `QuarkusTransaction.joiningExisting()` invocations where applicable.

## Motivation for the change:
Ensures operations actually occur in a transactional context as expected

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... bash smoketest.bash...*
2. Click around UI performing actions and ensure general functionality works as expected. Pay particular attention to Topology view/actions, Archives > All Targets, and Automated Rules, since these exercise things the most.
3. Create an automated rule like the one below and ensure that it correctly creates a new recording when enabled, that the recording gets archived every 10 seconds, and after there are 3 archived copies that the oldest one gets pruned when a new one is created. Disabling a rule should stop new archives from being saved.

```json
{
  "id": 1,
  "name": "test",
  "description": "",
  "matchExpression": "target.connectUrl.contains('localhost:0')",
  "eventSpecifier": "template=Profiling,type=TARGET",
  "archivalPeriodSeconds": 10,
  "initialDelaySeconds": 0,
  "preservedArchives": 3,
  "maxAgeSeconds": 300,
  "maxSizeBytes": 0,
  "enabled": true
}
```
